src/core/internal/abort.d

```diff
@@ -0,0 +1,44 @@
+module core.internal.abort;
+
+/*
+ * Use instead of assert(0, msg), since this does not print a message for -release compiled
+ * code, and druntime is -release compiled.
+ */
+void abort(string msg, string filename = __FILE__, size_t line = __LINE__) @nogc nothrow @safe
+{
+    // use available OS system calls to print the message to stderr
+    version(Posix)
+    {
+        import core.sys.posix.unistd: write;
+        static void writeStr(const(char)[][] m...) @nogc nothrow @trusted
+        {
+            foreach(s; m)
+                cast(void)write(2, s.ptr, s.length);
```

Comment:
Heh, I original thought to make abort pure (for more utility), but realized later that it cannot be made pure, because it would be considered a strong pure function without any return.

However, while I did have it pure, I had to "fake" write being pure. The compiler complained I wasn't using the output, and so I followed the instructions (put a cast(void) there ;)

It's a relic, will remove.